users/api/http/transport.go

```diff
@@ -105,28 +102,34 @@ func encodeResponse(_ context.Context, w http.ResponseWriter, response interface
 		if ar.Empty() {
 			return nil
 		}
+
+		w.Header().Set("Content-Type", contentType)
 	}
 
 	return json.NewEncoder(w).Encode(response)
 }
 
 func encodeError(_ context.Context, err error, w http.ResponseWriter) {
-	w.Header().Set("Content-Type", contentType)
-
-	switch err {
-	case users.ErrMalformedEntity:
-		w.WriteHeader(http.StatusBadRequest)
-	case users.ErrUnauthorizedAccess:
-		w.WriteHeader(http.StatusForbidden)
-	case users.ErrConflict:
-		w.WriteHeader(http.StatusConflict)
-	case errUnsupportedContentType:
-		w.WriteHeader(http.StatusUnsupportedMediaType)
-	case io.ErrUnexpectedEOF:
-		w.WriteHeader(http.StatusBadRequest)
-	case io.EOF:
-		w.WriteHeader(http.StatusBadRequest)
-	default:
+	// For debug only:
+	mfErr, ok := err.(errors.Error)
+	if ok {
+		switch {
+		case mfErr.Contains(users.ErrMalformedEntity):
+			w.WriteHeader(http.StatusBadRequest)
+			logger.Warn(fmt.Sprintf("Failed to decode user credentials: %s", err))
+		case mfErr.Contains(users.ErrUnauthorizedAccess):
+			w.WriteHeader(http.StatusForbidden)
+		case mfErr.Contains(users.ErrConflict):
+			w.WriteHeader(http.StatusConflict)
+		case mfErr.Contains(ErrUnsupportedContentType):
+			w.WriteHeader(http.StatusUnsupportedMediaType)
+			logger.Warn("Invalid or missing content type.")
+		case mfErr.Contains(io.ErrUnexpectedEOF):
+			w.WriteHeader(http.StatusBadRequest)
+		case mfErr.Contains(io.EOF):
+			w.WriteHeader(http.StatusBadRequest)
+		}
+	} else {
```

Comment:
This about how to avoid this `else`.

Reply:
I manage to avoid it with to switch, please check again. 
Also, I think that json error types are not possible anymore, I keep them from earlier version. But if you agree, I would remove that two cases.

Reply:
I removed json errors from type switch, I'm pretty sure they are not possible any more